File: Player.java
/****************************************************************************
 * Created by: Shuvaethy Neill
 * Created on: May 2019
 * Created for: ICS4U
 * This program is the game of 21
 ****************************************************************************/

public class Player
{
	//deal cards (2)
	public void DealNewCards(Card[] cardsToDeal) {
		for (int idx = 0; idx < cardsToDeal.length; idx++) {
			dealtCards[idx] = cardsToDeal[idx];
		}
	}

	//deal third card
	public void DealThirdCard(Card thirdCard) {
		dealtCards[2] = thirdCard;
	}

	public int GetSumOfAllDealtCards() {
		int sumOfDealtCardValues = 0;
		boolean seenAce = false;
		for (int i = 0; i < dealtCards.length; i++) {
			if (dealtCards[i] == null) {
				continue;
			}

			if (dealtCards[i].name.charAt(0) == 'A' && seenAce == false) {
				seenAce = true;
			}
			//add all dealt card values
			sumOfDealtCardValues += dealtCards[i].value;	
		}

		if (sumOfDealtCardValues <= 11 && seenAce == true) {
			sumOfDealtCardValues = sumOfDealtCardValues + 10;
		}

		//return sum
		return sumOfDealtCardValues;
	}

	//public String playerType;
	public Card[] dealtCards = new Card[3];

	//shows cards dealt
	public String toString() {
		String dealtCardsSummary = "";
		for (int i = 0; i < dealtCards.length; i++) {
			if (dealtCards[i] == null) {
				continue;
			}
			dealtCardsSummary += "Card #" + (i + 1) + "\n" + dealtCards[i] + "\n";
		}

		return dealtCardsSummary;
	}
}

File: player.py
class Player:

  def __init__(self):
      self.dealtCards = [None] * 3

  #//deal cards (2)
  def dealNewCards(self, cardsToDeal):
	  for idx in range(len(cardsToDeal)):
			self.dealtCards[idx] = cardsToDeal[idx]

	#//deal third card
  def dealThirdCard(self, thirdCard):
    self.dealtCards[2] = thirdCard
    
  def getSumOfAllDealtCards(self): 
		__sumOfDealtCardValues = 0
		seenAce = False;

		for i in range (len(self.dealtCards)):
			if self.dealtCards[i] == None:
				continue

			if (self.dealtCards[i].isAce() and seenAce == False):
				seenAce = True
			# add all dealt card values
			__sumOfDealtCardValues += self.dealtCards[i].getValue()

		if __sumOfDealtCardValues <= 11 and seenAce == True:
			__sumOfDealtCardValues = __sumOfDealtCardValues + 10

		#//returns sum
		return __sumOfDealtCardValues

	#//shows cards dealt
  def toString(self): 
		dealtCardsSummary = "\n"
		for i in range(len(self.dealtCards)):
			if self.dealtCards[i] == None:
				break

			dealtCardsSummary += "Card #" + str(i + 1) + "\n" + self.dealtCards[i].toString() + "\n"
		return dealtCardsSummary
